stk/calculators/energy/energy_calculators.py

```diff
@@ -377,3 +382,887 @@ def energy(self, mol, conformer=-1):
+
+        charge : :class:`int`, optional
+            Formal molecular charge.
+
+        unpaired_electrons : :class:`int`, optional
+            Number of unpaired electrons.
+
+        use_cache : :class:`bool`, optional
+            If ``True`` :meth:`energy` will not run twice on the same
+            molecule and conformer.
+
+        unlimited_memory : :class: `bool`, optional
+            If ``True`` :meth:`energy` will be run without constraints
+            on the stacksize. If memory issues are encountered, this
+            should be ``True``, however this may raise issues on
+            clusters.
+
+        """
+        self.xtb_path = xtb_path
+        self.gfn_version = gfn_version
+        self.output_dir = output_dir
+        self.num_cores = str(num_cores)
+        self.electronic_temperature = str(electronic_temperature)
+        self.solvent = solvent
+        if self.solvent is not None:
+            self.solvent = solvent.lower()
+            valid_xtb_solvent(
+                gfn_version=self.gfn_version,
+                solvent=self.solvent
+            )
+        self.solvent_grid = solvent_grid
+        self.charge = str(charge)
+        self.unpaired_electrons = str(unpaired_electrons)
+        self.unlimited_memory = unlimited_memory
+
+        self.total_energies = {}
+        self.homo_lumo_gaps = {}
+        self.fermi_levels = {}
+        self.homo_lumo_orbitals = {}
+        self.qonly_dipole_moms = {}
+        self.full_dipole_moms = {}
+        self.qonly_quadrupole_moms = {}
+        self.qdip_quadrupole_moms = {}
+        self.full_quadrupole_moms = {}
+        super().__init__(use_cache=use_cache)
+
+    def _get_properties(self, mol, conformer, output_file):
+        """
+        Extracts properties from GFN-xTB energy calculation.
+
+        Parameters
+        ----------
+        mol : :class:`.Molecule`
+            The :class:`.Molecule` whose energy is to be calculated.
+
+        conformer : :class:`int`, optinal
+            The conformer of `mol` to use.
+
+        output_file : :class: `str`
+            Name of output file with xTB results.
+
+        Returns
+        -------
+        None : :class:`NoneType`
+
+        """
+        xtbext = XTBExtrators(output_file=output_file)
+
+        # Get properties from output file.
+        key = (mol, conformer)
+        self.total_energies[key] = xtbext.total_energy()
+        self.homo_lumo_gaps[key] = xtbext.homo_lumo_gap()
+        self.fermi_levels[key] = xtbext.fermi_level()
+        self.homo_lumo_orbitals[key] = xtbext.homo_lumo_occ()
+        self.qonly_dipole_moms[key] = xtbext.qonly_dipole_mom()
+        self.full_dipole_moms[key] = xtbext.full_dipole_mom()
+        self.qonly_quadrupole_moms[key] = xtbext.qonly_quadrupole_mom()
+        self.qdip_quadrupole_moms[key] = xtbext.qdip_quadrupole_mom()
+        self.full_quadrupole_moms[key] = xtbext.full_quadrupole_mom()
+
+    def _write_and_run_command(self, mol, conformer):
```

Comment:
I think if your function has the word "and" in its name, it's a good indicator that two functions should be used in succession instead.

Reply:
this is a good point. I try to

Reply:
Normally I would agree, however, I think these two processes are so inherently linked and to require two functions just adds function calls and doesn't improve the readability of the code. If you disagree, then I can split them, but for now I am leaving them as is.

Reply:
I dont think the processes are inherently linked. I think they are only linked by the conext of  their calling funciton. IE writing does not depending on running the command. The execution of a subprocess call is also independent of the write being done first. Now of course, in the context of an energy calculation the write needs to happen first, followed by a subprocess call, which is why the energ() function is the thing that links these two functions within a context.

```python

mol.write(...)
self._run_cmd()

``` 

reads better.

Reply:
Also write_and_run_cmd() raises the question "write what?" whereas this makes it clear that the mol is getting written. I would say its cool to put mol.write() within _run_cmd, but just call it _run_cmd() dont call it write_and_run_cmd()

Reply:
I agree with this and realized the confusion due to how I wrote the code.